### PR TITLE
Raven sentry ( Not for egest )

### PR DESCRIPTION
**Pull request by akilananb**
Added Error reporting for pipeline !( Not for egest )

### CODE REVIEW

File: server/pipelines/v3.js [hunk 1]
```diff
@@ -174,6 +176,7 @@ Pipeline.prototype.process = (message) => {
 	})
 	.catch(error => {
 		console.log('pipeline error', error);
+	    report_error.capture('pipeline annotation error',error);
```
Comment (akilananb) [inline server/pipelines/v3.js]: I don't know why its formatted incorrectly, For me it looks pretty in my machine....What IDE and spacing is advised
...?

File: server/report-error.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import raven from 'raven';
```
Comment (commuterjoy) [inline server/report-error.js]: can you put this file in ./server/utils please ?

File: server/report-error.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+let client;
+if(!client) 
+{
+	client= new raven.Client(process.env.RAVEN_CONNECTION_URI);
```
Comment (commuterjoy) [inline server/report-error.js]: Can you add another environment variable, say, `RAVEN_FLAG` that allows us to turn Raven on and off?

File: server/transforms/device-atlas.js [hunk 4]
```diff
@@ -43,9 +45,10 @@ module.exports = function (event) {
 
 			const [status, res] = response;
 
-			console.log('transforms/device-atlas', ua, status, JSON.stringify(res));
+			console.log('transforms/device-atlas', ua, status, JSON.stringify(res),true);
```
Comment (commuterjoy) [inline server/transforms/device-atlas.js]: `true` ?

File: server/transforms/is-staff.js [hunk 5]
```diff
@@ -1,9 +1,11 @@
 import metrics from 'next-metrics';
+import report_error from '../report-error';
 
 module.exports = function (event) {
 	return new Promise((resolve, reject) => {
 		
 		if (!process.env.transform_is_staff && !process.env.mocha) {
+			report_error.captureMessage('Transform is_staff is switched off',{level:'info'});
```
Comment (commuterjoy) [inline server/transforms/is-staff.js]: these sort of things aren't errors, we may want to turn a transform off for a valid reason

File: server/transforms/session-api.js [hunk 6]
```diff
@@ -43,6 +45,7 @@ module.exports = (event) => {
 				if (res.status === 200) {
 					return res.json();
 				} else {
+            		report_error.captureMessage('pipeline.transforms.sessionApi.error'+JSON.stringify(res),{level:'debug'});
```
Comment (commuterjoy) [inline server/transforms/session-api.js]: I don't think we should report non-200 responses to sentry as, i) there will probably be a lot of them, ii) they are expected, and we handle them.

Sentry is mostly of interest for uncaught exceptions and errors.

File: server/pipelines/v3.js (hunk 1 above)
Comment (commuterjoy) [inline server/pipelines/v3.js]: We generally use tabs, although I've not got http://eslint.org/ running on this project, so perhaps we should do that!

**Comment by commuterjoy** [comment]
Also, there's a couple of try/catch statements in this file - can you add the error reporting to the errors please?

https://github.com/Financial-Times/spoor-enrichment/blob/master/server/models/event.js#L21

File: server/enrichment.js [hunk 7]
```diff
@@ -4,7 +4,7 @@ import AWS from 'aws-sdk';
 import Pipeline	from './pipelines';
 import metrics from 'next-metrics';
 import sinks from './sinks';
-
+import report_error from './utils/report-error'
```
Comment (commuterjoy) [inline server/enrichment.js]: Stylistically we use camel case, Eg, reportError

File: server/transforms/time.js [hunk 8]
```diff
@@ -1,6 +1,7 @@
 
 import moment from 'moment';
 import metrics from 'next-metrics';
+import report_error from '../utils/report-error';
```
Comment (commuterjoy) [inline server/transforms/time.js]: probably can remove this if it's not used in the module

File: server/enrichment.js (hunk 7 above)
Comment (akilananb) [inline server/enrichment.js]: no problem i'll update it

**Comment by commuterjoy** [comment]
+1 looks good

**Pull request merged**